### DISCUSSION TITLE
ACPI-CA

### DISCUSSION THREAD

**Discussion by hyperswine** [rust-osdev/acpi#118] [Ideas]
Any plans to explore ACPI component arch? Would be great to have something in rust that could access ACPI functionality/interpret AML bytecode in a cleaner way.

Functionalities im [talking about ](https://people.freebsd.org/~takawata/acpipaper/acpi_freenix/node8.html)

**Comment by IsaacWoods** [comment]
One of the key reasons for this library implementing AML from the ground up, rather than for example wrapping ACPICA, is to explore how we'd approach ACPI in a modern language. I personally think ACPICA's code is very difficult to approach, and that the API is basically impossible to use correctly without an in-depth knowledge of ACPI.

Why do you think emulating the ACPICA interface would be cleaner?
